cpp/include/cuspatial/traj_thrust.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <ostream>
+#include <cuspatial/cuspatial.h>
+
+namespace cuspatial
+{
+
+	/**
+	 *@brief Thrust functor for comparing two TimeStamp variables; used in sorting based on timestamp
+	 */
+	__host__ __device__
```

Comment:
Do you ever call this from the host? If not, make it device-only.

```suggestion
	__device__
```

Reply:
But it can hurt. See my section on __host__ __device__ in the C++ code standard I shared with you. Add it if and when we need it. Not "just in case".